src/modules/src/crtp_localization_service.c

```diff
@@ -363,31 +390,107 @@ void locSrvSendRangeFloat(uint8_t id, float range)
+  matchedAnglePacket *packetData = (matchedAnglePacket *)packet->data;
+
+  packetData->baseStation = baseStation;
+  packetData->group_id_and_bs_count = 0; // will be set later when sending the packet
+
+  const pulseProcessorBaseStationMeasurement_t* baseStationMeasurement = &angles->baseStationMeasurementsLh1[baseStation];
+  for(uint8_t its = 0; its < NBR_OF_SWEEPS_IN_PACKET; its++) {
+    float angle_first_sensor =  baseStationMeasurement->sensorMeasurements[0].correctedAngles[its];
+    packetData->sweeps[its].sweep = angle_first_sensor;
+
+    for(uint8_t itd = 0; itd < NBR_OF_SENSOR_DIFFS_IN_PACKET; itd++) {
+      float angle_other_sensor = baseStationMeasurement->sensorMeasurements[itd + 1].correctedAngles[its];
+      uint16_t angle_diff = single2half(angle_first_sensor - angle_other_sensor);
+      packetData->sweeps[its].angleDiffs[itd].angleDiff = angle_diff;
+    }
+  }
 
-    for(uint8_t its = 0; its < NBR_OF_SWEEPS_IN_PACKET; its++) {
-      float angle_first_sensor =  baseStationMeasurement->sensorMeasurements[0].correctedAngles[its];
-      ap->sweeps[its].sweep = angle_first_sensor;
+  packetData->type = LH_MATCHED_ANGLE_STREAM;
+  lhMatchedAnglePackets[lhMatchedStreamBsCount].port = CRTP_PORT_LOCALIZATION;
+  lhMatchedAnglePackets[lhMatchedStreamBsCount].channel = GENERIC_TYPE;
+  lhMatchedAnglePackets[lhMatchedStreamBsCount].size = sizeof(matchedAnglePacket);
+}
 
-      for(uint8_t itd = 0; itd < NBR_OF_SENSOR_DIFFS_IN_PACKET; itd++) {
-        float angle_other_sensor = baseStationMeasurement->sensorMeasurements[itd + 1].correctedAngles[its];
-        uint16_t angle_diff = single2half(angle_first_sensor - angle_other_sensor);
-        ap->sweeps[its].angleDiffs[itd].angleDiff = angle_diff;
+static void locSrvSendLighthouseAngleMatchedBs(const int baseStation, const pulseProcessorResult_t* angles, const uint32_t now_ms)
+{
+  // Check if we past the end time of the group, send the previous batch of packets and reset
+  if (now_ms > lhMatchedStreamGroupEndtime) {
+    if (lhMatchedStreamBsCount >= lhMatchedStreamMinBsCount) {
+      // Check that the data to send is not too old.
+      // This is an edge case if there is a full group just before no more LH data is received.
+      // When the reception is restored we don't want to send the old data.
+      const uint32_t age = now_ms - lhMatchedStreamGroupEndtime;
+      if (age < LH_MATCHED_STREAM_MAX_PENDING_TIME_MS) {
+        for (int i = 0; i < lhMatchedStreamBsCount; i++) {
+          matchedAnglePacket *ap = (matchedAnglePacket *)lhMatchedAnglePackets[i].data;
+          ap->group_id_and_bs_count = (lhMatchedStreamGroupId << 4) | (lhMatchedStreamBsCount & 0x0F);
+          // This is best effort, i.e. the blocking version is not needed
+          crtpSendPacket(&lhMatchedAnglePackets[i]);
+        }
+
+        lhMatchedStreamGroupId++;
+        if (lhMatchedStreamGroupCount > 0 && lhMatchedStreamGroupCount != 255) {
+          lhMatchedStreamGroupCount--;
```

Comment:
This could be implemented using the `paramSetInt()` function : https://github.com/bitcraze/crazyflie-firmware/blob/03adf38b3bc1cae7f22e0cb7c62320f0f43b014a/src/modules/interface/param_logic.h#L117-L124.

This would send a packet to the ground each time the value is modified. If that is not detrimental to the usecase it would be preferable. If so many downlink packets is a problem though, I would suggest at least sending a notification when the value reaches 0.

Reply:
The main goal of this new functionality was to reduce the number of packets that are sent so I was a bit reluctant to update the parameter. Doing it once when reaching the end seems as a good compromise though, I'll do that.